src/components/graph/ExerciseTreeGraph.tsx

```diff
@@ -295,6 +298,64 @@ export class ExerciseTreeGraph extends React.Component<ExerciseTreeGraphProps> {
   }
 }
 
+function drawNode(
+  nodeSelection: d3.Selection<
```

Comment:
You might want to extract this type (here or future PR)

Reply:
The generic types sometimes change with the d3.Selection type, I can't find good documentation how to predict what the generic types will be.

Reply:
```typescript
type DrawableNodeSelection = d3.Selection<
  SVGGElement,
  ExerciseGraphNode,
  SVGGElement,
  unknown
>
```

This not worky-work?

Reply:
I'll see if I can find different variations I've seen

Reply:
I am dumb.  I went to look at the types as they are defined in `@types/d3-selection`, someone did a great job:

```
/**
 * A D3 Selection of elements.
 *
 * The first generic "GElement" refers to the type of the selected element(s).
 * The second generic "Datum" refers to the type of the datum of a selected element(s).
 * The third generic "PElement" refers to the type of the parent element(s) in the D3 selection.
 * The fourth generic "PDatum" refers to the type of the datum of the parent element(s).
 */
```

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/d3-selection/index.d.ts#L171-178

Reply:
I can extract the types predictably now.  :zipper_mouth_face: